CMakePresets.json

```diff
@@ -6,9 +6,69 @@
     "patch": 0
   },
   "configurePresets": [
+    {
+      "name": "base",
+      "description": "Base options",
+      "binaryDir": "${sourceDir}/build",
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INSTALL_PREFIX": "${sourceDir}/install",
+        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "OFF",
+        "FULL_BUILD": "ON"
```

Comment:
This is good for CI. But if we want to use presets for development, we should probably change it. But it is fine for the purposes of this PR.

Reply:
We'd have different presets for development.